CRM/Mailchimptemplate/Form/MailchimpTemplateSettings.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+require_once 'CRM/Core/Form.php';
+require_once 'vendor/MailChimp.php';
+
+use CRM_Mailchimptemplate_ExtensionUtil as E;
+
+/**
+ * Form controller class
+ *
+ * @see http://wiki.civicrm.org/confluence/display/CRMDOC43/QuickForm+Reference
+ */
+class CRM_Mailchimptemplate_Form_MailchimpTemplateSettings extends CRM_Core_Form
+{
+    private $apikey = null;
+
+    public function preProcess()
+    {
+        parent::preProcess();
+        $this->apikey = CRM_Mailchimptemplate_Settings::getApikey();
+    }
+
+    public function buildQuickForm()
+    {
+        // add form elements
+        $this->add(
+            'text', // field type
+            CRM_Mailchimptemplate_Settings::ELFIELDNAME, // field name
+            E::ts('MailChimp API key'), // field label
+            null, // list of options
+            false // is required
+        )
+            ->setValue($this->apikey);
+
+        $this->addButtons(
+            array(
+                array(
+                    'type' => 'submit',
+                    'name' => E::ts('Submit'),
+                    'isDefault' => true,
+                ),
+            )
+        );
+
+        // export form elements
+        $this->assign('elementNames', $this->getRenderableElementNames());
+        parent::buildQuickForm();
+    }
+
+    public function postProcess()
+    {
+        CRM_Mailchimptemplate_Settings::setApikey($this->exportValue(CRM_Mailchimptemplate_Settings::ELFIELDNAME));
```

Comment:
This value should be sanitized as it is user input
`CRM_RcBase_Processor_Base::sanitizeString()`

Reply:
Docs says exportvalue returns elements' safe value. That's why I didn't sanitize.
https://pear.php.net/manual/en/package.html.html-quickform.html-quickform.exportvalue.php

Reply:
I think it will only sanitize if you register validation and sanitization rules before.
`<script>alert('evil')</script>` was saved w/o a problem

Reply:
Yes, that code was saved but doesn't get executed. Anyway I made your suggestion. There is some encoding when saving to the database (https://docs.civicrm.org/dev/en/latest/security/inputs/#when-saving-to-the-database) but the input value encrypted before saving so I don't know where is the encoding happens.